digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/AwsService.scala

```diff
@@ -0,0 +1,299 @@
+          awsDao.updateFeatureStatus(featureId, "6")
+          awsDao.updateDatasetStatus(datasetId, 2)
+        }
+    }
+  }
+
+  private def updatePoint(properties: Map[String, Any], link: Option[RoadLink], assetType: String, assetCoordinates: List[Double]): Unit = {
+    assetType match {
+      case "obstacle" =>
+        val obstacleType = properties("class").asInstanceOf[BigInt]
+        val newObstacle = IncomingObstacle(assetCoordinates.head, assetCoordinates(1), link.get.linkId, obstacleType.intValue())
+
+        obstacleService.checkDuplicates(newObstacle, true) match {
+          case Some(value) =>
+            obstacleService.expire(value.id, AwsUser)
+            obstacleService.create(newObstacle, AwsUser, link.get)
+          case None =>
+            obstacleService.create(newObstacle, AwsUser, link.get)
+        }
+    }
+  }
+
+  private def updateRoadlink(properties: Map[String, Any], links: Seq[RoadLink]) = {
+    val speedLimit = properties.getOrElse("speedLimit", "").asInstanceOf[String]
+    val pavementClass = properties.getOrElse("pavementClass", "").asInstanceOf[String]
+
+    links.foreach { link =>
+      speedLimit match {
+        case "" =>
+        case _ =>
+          speedLimitService.getExistingAssetByRoadLink(link) match {
+            case Some(value) =>
+              val newSpeedLimitAsset = NewLinearAsset(link.linkId, value.startMeasure, value.endMeasure, NumericValue(speedLimit.toInt), properties("sideCode").asInstanceOf[BigInt].intValue(), vvhClient.roadLinkData.createVVHTimeStamp(), None)
+              speedLimitService.update(value.id, Seq(newSpeedLimitAsset), AwsUser)
+            case None =>
+              val newSpeedLimitAsset = NewLinearAsset(link.linkId, 0, link.length, NumericValue(speedLimit.toInt), properties("sideCode").asInstanceOf[BigInt].intValue(), vvhClient.roadLinkData.createVVHTimeStamp(), None)
+              speedLimitService.createMultiple(Seq(newSpeedLimitAsset), speedLimit.toInt, AwsUser, vvhClient.roadLinkData.createVVHTimeStamp(), (_, _) => Unit)
+          }
+      }
+
+      pavementClass match {
+        case "" =>
+        case _ =>
+          val pavementClassValue = DynamicValue(DynamicAssetValue(Seq(DynamicProperty("paallysteluokka", "single_choice", required = false, Seq(DynamicPropertyValue(pavementClass.toInt))))))
+
+          val duplicate = pavedRoadService.getPersistedAssetsByLinkIds(PavedRoad.typeId, Seq(link.linkId))
+          duplicate.isEmpty match {
+            case false =>
+              pavedRoadService.update(Seq(duplicate.head.id), pavementClassValue, AwsUser)
+            case true =>
+              val newPavementClassAsset = NewLinearAsset(link.linkId, 0, link.length, pavementClassValue, properties("sideCode").asInstanceOf[BigInt].intValue(), vvhClient.roadLinkData.createVVHTimeStamp(), None)
+              pavedRoadService.create(Seq(newPavementClassAsset), PavedRoad.typeId, AwsUser)
+          }
+      }
+    }
+  }
```

Comment:
It is better if we use get instead of getOrElse in the first 2 lines and then use match with options fetched in those 2 lines